metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/NotificationEventProducer.java

```diff
@@ -1,62 +1,30 @@
 package org.hypertrace.alert.engine.metric.anomaly.detector;
 
 import com.typesafe.config.Config;
-import java.io.IOException;
-import java.nio.ByteBuffer;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Properties;
 import org.apache.kafka.clients.producer.KafkaProducer;
 import org.apache.kafka.clients.producer.Producer;
-import org.apache.kafka.clients.producer.ProducerConfig;
 import org.apache.kafka.clients.producer.ProducerRecord;
 import org.hypertrace.alert.engine.metric.anomaly.datamodel.MetricAnomalyNotificationEvent;
 import org.hypertrace.alert.engine.metric.anomaly.datamodel.queue.KafkaConfigReader;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 class NotificationEventProducer {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(MetricAnomalyDetector.class);
-
-  private final Producer<String, ByteBuffer> producer;
-  private final KafkaConfigReader kafkaConfigReader;
+  private final Producer<String, MetricAnomalyNotificationEvent> producer;
   private final Properties properties;
 
   NotificationEventProducer(Config kafkaQueueConfig) {
-    this.kafkaConfigReader = new KafkaConfigReader(kafkaQueueConfig);
+    KafkaConfigReader kafkaConfigReader = new KafkaConfigReader(kafkaQueueConfig);
     this.properties = new Properties();
-    properties.putAll(kafkaConfigReader.getProducerConfig(createBaseProperties()));
-    producer = new KafkaProducer<String, ByteBuffer>(properties);
+    properties.putAll(kafkaConfigReader.getProducerConfig());
+    producer = new KafkaProducer<>(properties);
   }
 
   public void publish(MetricAnomalyNotificationEvent notificationEvent) {
-    try {
-      producer.send(
-          new ProducerRecord<String, ByteBuffer>(
-              properties.getProperty("topic"), null, notificationEvent.toByteBuffer()));
-    } catch (IOException e) {
-      LOGGER.error("Exception producing messages", e);
-    }
+    producer.send(new ProducerRecord<>(properties.getProperty("topic"), null, notificationEvent));
```

Comment:
As a follow-up, we may have to hook ErrorHandler in properties to handle serde errors if any.